File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Leisure.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Leisure

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _eatingOut As Decimal
    <DisplayName("Eating Out"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property eatingOut() As Decimal
        Get
            Return _eatingOut
        End Get
        Set(ByVal value As Decimal)
            _eatingOut = value
            totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol
        End Set
    End Property

    Private _cinema As Decimal
    <DisplayName("Cinema"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property cinema() As Decimal
        Get
            Return _cinema
        End Get
        Set(ByVal value As Decimal)
            _cinema = value
            totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol
        End Set
    End Property

    Private _holidays As Decimal
    <DisplayName("Holidays"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property holidays() As Decimal
        Get
            Return _holidays
        End Get
        Set(ByVal value As Decimal)
            _holidays = value
            totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol
        End Set
    End Property

    Private _sports As Decimal
    <DisplayName("Sports"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property sports() As Decimal
        Get
            Return _sports
        End Get
        Set(ByVal value As Decimal)
            _sports = value
            totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol
        End Set
    End Property

    Private _cigarettesAndAlcohol As Decimal
    <DisplayName("Cigarettes/Alcohol"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property cigarettesAndAlcohol() As Decimal
        Get
            Return _cigarettesAndAlcohol
        End Get
        Set(ByVal value As Decimal)
            _cigarettesAndAlcohol = value
            totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol
        End Set
    End Property

    Private _totalLeisure As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Leisure Outgoings")> _
    Public Property totalLeisure() As Decimal
        Get
            Return _totalLeisure
        End Get
        Set(ByVal value As Decimal)
            _totalLeisure = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Leisure", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalLeisure.ToString("c2")
    End Function

End Class

File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Leisure.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;

namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Leisure
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _eatingOut;
        [DisplayName("Eating Out"), RefreshProperties(RefreshProperties.All)]
        public decimal eatingOut
        {
            get { return _eatingOut; }
            set
            {
                _eatingOut = value;
                totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol;
            }
        }

        private decimal _cinema;
        [DisplayName("Cinema"), RefreshProperties(RefreshProperties.All)]
        public decimal cinema
        {
            get { return _cinema; }
            set
            {
                _cinema = value;
                totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol;
            }
        }

        private decimal _holidays;
        [DisplayName("Holidays"), RefreshProperties(RefreshProperties.All)]
        public decimal holidays
        {
            get { return _holidays; }
            set
            {
                _holidays = value;
                totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol;
            }
        }

        private decimal _sports;
        [DisplayName("Sports"), RefreshProperties(RefreshProperties.All)]
        public decimal sports
        {
            get { return _sports; }
            set
            {
                _sports = value;
                totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol;
            }
        }

        private decimal _cigarettesAndAlcohol;
        [DisplayName("Cigarettes/Alcohol"), RefreshProperties(RefreshProperties.All)]
        public decimal cigarettesAndAlcohol
        {
            get { return _cigarettesAndAlcohol; }
            set
            {
                _cigarettesAndAlcohol = value;
                totalLeisure = eatingOut + cinema + holidays + sports + cigarettesAndAlcohol;
            }
        }

        private decimal _totalLeisure;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Leisure Outgoings")]
        public decimal totalLeisure
        {
            get { return _totalLeisure; }
            set
            {
                _totalLeisure = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Leisure",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalLeisure.ToString("c2");
        }


    }
}
